File: dropwizard-scala_2.9.1/src/main/scala/com/yammer/dropwizard/bundles/ScalaBundle.scala
package com.yammer.dropwizard.bundles

import com.yammer.dropwizard.Bundle
import com.yammer.dropwizard.config.Environment
import com.yammer.dropwizard.providers.OauthTokenProvider
import com.codahale.jersey.inject.ScalaCollectionsQueryParamInjectableProvider
import com.codahale.jersey.providers.JerksonProvider

object ScalaBundle extends Bundle {
  def initialize(environment: Environment) {
    environment.addProvider(new JerksonProvider[Any])
    environment.addProvider(new OauthTokenProvider)
    environment.addProvider(new ScalaCollectionsQueryParamInjectableProvider)
  }
}

File: dropwizard-scala_2.9.1/src/main/java/com/yammer/dropwizard/bundles/ScalaBundle.java
package com.yammer.dropwizard.bundles;

import com.fasterxml.jackson.module.scala.DefaultScalaModule;
import com.yammer.dropwizard.Bundle;
import com.yammer.dropwizard.config.Bootstrap;
import com.yammer.dropwizard.config.Environment;
import com.yammer.dropwizard.scala.inject.ScalaCollectionsQueryParamInjectableProvider;

public class ScalaBundle implements Bundle {
    @Override
    public void initialize(Bootstrap<?> bootstrap) {
        bootstrap.getObjectMapperFactory().registerModule(new DefaultScalaModule());
    }

    @Override
    public void run(Environment environment) {
        environment.addProvider(new ScalaCollectionsQueryParamInjectableProvider());
    }
}
